### ISSUE TITLE
[Bug] fix tp_size > kv nums head crash

### ISSUE THREAD

**Issue by Prayer3th** [sgl-project/sglang-jax#199]
### Checklist

- [x] 1. I have searched related issues but cannot get the expected help.
- [x] 2. The bug has not been fixed in the latest version.
- [x] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [x] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sgl-jax/discussions/new/choose Otherwise, it will be closed.
- [x] 5. Please use English, otherwise it will be closed.

### Describe the bug

fix tp_size > kv nums head crash

### Reproduction

JAX_COMPILATION_CACHE_DIR=/tmp/jit_cache  python3 -u -m sgl_jax.launch_server --model-path Qwen/Qwen3-30B-A3B --trust-remote-code  --dist-init-addr=0.0.0.0:10011 --nnodes=1  --tp-size=8 --device=tpu --random-seed=3 --node-rank=0 --mem-fraction-static=0.85 --chunked-prefill-size=2048 --download-dir=/tmp --dtype=bfloat16 --max-running-requests 256 --skip-server-warmup --page-size=128 --disable-radix-cache --precompile-bs-paddings 1 --precompile-token-paddings 512

### Environment

v6e * 8
Qwen3-30B-A3B
